src/api/index.ts

```diff
@@ -170,7 +163,26 @@ export class GithubDeploymentsApiClient implements GithubDeploymentsApi {
       return formattedStatues;
     };
 
-    const formatted = formatStatuses(statuses);
-    return formatted;
+    const page = 100;
+    const pages = Math.ceil(params.deploymentNodeIds.length / page);
+    const statusRequests = [];
+    const formattedStatues: RestDeploymentStatus[] = [];
+
+    for (var i = 0; i < pages; i++) {
+      const sliced = params.deploymentNodeIds.slice(i * page, (i + 1) * page);
+      statusRequests.push(
+        await octokit<statusesResponse>(statusesQuery, {
+          deploymentNodeIds: sliced,
+        }),
+      );
+    }
+
+    await Promise.all(statusRequests).then(response => {
```

Comment:
If Brady's team already has 100+ deployments in this one repo, are we worried the total number of deployments in a repo could get to the point where the TechHub deployments page takes a long time to load? Would it make sense to optimize this and return the pages as they come back instead of doing `Promise.all`? Or do you think that is an unnecessary optimization at this point?